docs/proposals/dbaas/01-architecture/proposal.md

```diff
@@ -0,0 +1,695 @@
+
+#### Story 11
+As an SRE person, I should be able to configure storage for backups.
+
+#### Story 12
+As a user, I should be able to deploy a database with the recommended defaults
+
+#### Story 13
+As a user, I should be able to deploy a database with the selected version or cluster size
+
+#### Story 14
+As a user, I should be able to select a resource template to deploy a database
+
+#### Story 15
+As a user, I should be able to select a resource template and database engine configuration template to deploy a database
+
+#### Story 16
+As a user, I should be able to edit a database cluster (If I have sufficient permissions)
+
+#### Story 17
+As a user, I should be able to delete a database cluster (If I have sufficient permissions)
+
+#### Story 18
+As a user, I should be able to create a database cluster from a provided backup file.
+
+#### Story 19
+As a DBA, I should be able to tune performance for a database cluster.
+
+#### Story 20
+As a DBA(?), I should be able to view cluster resources available before creating a database.
+
+## Design Details
+
+### Air gapped environments
+The design does not support air gapped environments because of internet connectivity dependency.
+
+### OLM installation
+PMM always installs the latest version of OLM using the following process:
+
+1. Get the latest release from Github
+2. Download crds.yaml and olm.yaml
+3. Apply crds.yaml and olm.yaml against Kubernetes cluster
+
+### Installation of dbaas-catalog
+
+PMM uses [Percona DBaaS catalog](https://github.com/percona/dbaas-catalog/blob/main/docs/design/Percona-DBaaS-Platform-design.md). Percona DBaaS catalog is a repository of metadata that Operator Lifecycle Manager (OLM) can query to discover and install Operators and their dependencies on a cluster. OLM always installs Operators from the latest version of a catalog. It installs the following operators from this catalog:
+
+1. Percona XtraDB Cluster Operator
+2. Percona Server MongoDB Operator
+3. Victoria Metrics Operator
+4. DBaaS operator
+
+Any additional operators should be added to the catalog. Installation of an operator has the following steps:
+
+1. PMM should have DBaaS operator installed
+2. PMM creates [Operator Group](https://docs.openshift.com/container-platform/4.8/operators/understanding/olm/olm-understanding-operatorgroups.html) if it does not exist
+3. PMM creates [subscription](https://olm.operatorframework.io/docs/concepts/olm-architecture/) for the selected operator
+
+**Using OLM with pre-installed operators**
+
+1. PMM gets versions of installed operators
+2. PMM creates subscription with specified versions
+3. OLM installs [specific version](https://olm.operatorframework.io/docs/tasks/install-operator-with-olm/#example-install-a-specific-version-of-an-operator)
+4. Operator restarts and continues to work normally but now OLM manages installation of new versions
+
+
+### PMM REST API high-level design
```

Comment:
which REST API guidelines does this design follow? 
some endpoints uses `-`, the others `_`, some without separator at all